sda-commons-server-auth/src/test/java/org/sdase/commons/server/auth/key/RsaKeyLoaderSchedulerTest.java

```diff
@@ -0,0 +1,103 @@
+  private final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
+  private final RsaKeyLoaderScheduler keyLoaderScheduler =
+      RsaKeyLoaderScheduler.create(keyLoader, executorService);
+  private boolean elseCondition = false;
+
+  @Test
+  public void shouldNotBeAuthenticatedWithRevokedKey() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+    final RSAPublicKey keyToBeRevoked = Mockito.mock(RSAPublicKey.class);
+    final String kidForKeyToBeRevoked = "kidForKeyToBeRevoked";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return Arrays.asList(
+                  new LoadedPublicKey(kidForValidKey, validKey, this),
+                  new LoadedPublicKey(kidForKeyToBeRevoked, keyToBeRevoked, this));
+            } else {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.start(0, 2, TimeUnit.SECONDS);
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isSameAs(keyToBeRevoked);
+            });
+
+    // when -> simulate revoking of key:
+    elseCondition = true;
+
+    // then
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              // revoked key not found anymore
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isNull();
+            });
+  }
+
+  @Test
+  public void shouldContinueJobInCaseOfError() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            } else {
+              throw new KeyLoadFailedException("Some error occurred during reload");
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.start(0, 2, TimeUnit.SECONDS);
+    await().untilAsserted(() -> assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey));
+
+    // when -> simulate error-occurrence:
+    elseCondition = true;
+
+    // then -> key is found anyway:
+    await().untilAsserted(() -> assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey));
```

Comment:
Hmm. This is a tricky one. I am not sure if this really tests what it intends to do. We can't know for sure that the keysource was actually reloaded (it probably didn't).

I had a deeper look into the code (didn't do that on the phone in the morning) and figured that the `reloadKeys()` functionality is tested here: https://github.com/SDA-SE/sda-dropwizard-commons/blob/55762502ee95073157656a191b261ed69659d5f7/sda-commons-server-auth/src/test/java/org/sdase/commons/server/auth/key/RsaPublicKeyLoaderTest.java#L53-L83

Maybe this test case isn't as important as I thought. So you could go with one of the following options:
1. Make sure the test works as expected.
2. Remove the test (sorry for the inconvenience...).

At least a test to see that an exception did not break anything would be good.